core-spi-test/src/main/java/org/ehcache/internal/tier/CachingTierClear.java

```diff
@@ -103,7 +103,9 @@ public Object apply(final Object o) {
           }
         });
 
-        assertThat(newReturnedValueHolder.value(), is(equalTo(newValueHolder.value())));
+        if (newReturnedValueHolder != null) {
```

Comment:
No, this makes no sense with regards to the changes done here, same applies to the two other changed tests. None of these changes belong here.
Find the root cause of the failure and fix that instead.